src/commands/ServiceCommands.ts

```diff
@@ -189,6 +189,13 @@ async function connectLocalProject(service: LocalService): Promise<LocalProject>
   return localResourceExplorer.selectProject(await getExistingNames(service), await getExistingPorts(service), options);
 }
 
+// ------------ GENERIC ------------ //
+async function connectGenericProject(service: GenericService): Promise<GenericProject> {
+  const genericResourceExplorer = new GenericResourceExplorer();
+  return genericResourceExplorer.selectProject(await getExistingNames(service), await getExistingPorts(service));
```

Comment:
Question: is it good form to put awaits into method parameters? I know it works, just looks odd.

Reply:
no its ok, I just never saw that block of code before and just wondered. :-D